FreeRTOS/FreeRTOS
# Changes in FreeRTOS+TCP demo application files to add compatibility for running EndPoint changes in the IPv4 stack.

Changes in FreeRTOS+TCP demo application files to add compatibility for running EndPoint changes in the TCP stack.

Description
-----------
1. FreeRTOS_GetAddressConfiguration function is updated to **FreeRTOS_GetEndPointConfiguration** for supporting EndPoint changes
2. **FREERTOS_PLUS_TCP_VERSION** macro support is added to verify, which functionality to be used , with or without the EndPoint changes.
3. **vApplicationIPNetworkEventHook** function is added to UDP demo, to determine which demo project to run .
4.  Added **IPType** as parameter to **FreeRTOS_GetUDPPayloadBuffer** function, to be in sync with EndPoint changes 

Test Steps
-----------
Normal Demo run steps

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
